docs/celsius-converter-option/Snippets.re

```diff
@@ -0,0 +1,107 @@
+let _ = {
+  let celsius = "";
+  let convert = x => x;
+  // #region float-of-string-opt
+  switch (celsius |> float_of_string_opt) {
+  | None => "error"
+  | Some(fahrenheit) =>
+    (fahrenheit |> convert |> Js.Float.toFixedWithPrecision(~digits=2))
+    ++ {js|°F|js}
+  };
```

Comment:
I'm not completely satisfied with the snippets in this file because they end with `;`, which, if pasted directly into the user's code, will cause a syntax error. One workaround is to include the entire JSX expression, but that makes for overly-long code blocks.

I guess the way to handle it is to make it so files in docs directory can't be auto-formatted using `dune fmt` and then include a comment at the top of this file warning against hand-editing. Wdyt?

Reply:
Thought about this and my current plan is to split my snippets into two separate files: Snippets.re (for ones that can be auto-formatted), and NoFmtSnippets.re (for those that shouldn't be autoformatted). @jchavarri Wdyt?

Reply:
> if pasted directly into the user's code, will cause a syntax error.

Sorry, I am confused about this, could you elaborate? The way I see it, if they copy just the code inside the region, it's valid Reason snippet. If they copy the whole `let _ = {` (until the closing bracket) it's also valid syntax.

How will it cause a syntax error?

> Thought about this and my current plan is to split my snippets into two separate files: Snippets.re (for ones that can be auto-formatted), and NoFmtSnippets.re (for those that shouldn't be autoformatted).

This sounds quite convoluted to me.

Reply:
These snippets won't compile if they are pasted in because they're part of a larger expression which looks something like this:
```reason 
    {(
       String.trim(celsius) == ""
         ? {js|?°F|js}
         : (
           switch (celsius |> float_of_string_opt |> Option.map(convert)) {
           | None => "error"
           | Some(fahrenheit) when fahrenheit < (-128.6) => {js|Unreasonably cold🥶|js}
           | Some(fahrenheit) when fahrenheit > 212.0 => {js|Unreasonably hot🥵|js}
           | Some(fahrenheit) =>
             Js.Float.toFixedWithPrecision(fahrenheit, ~digits=2)
             ++ {js|°F|js}
           }
         )
     )
     |> React.string}
```
So the trailing `;` will break the expression when you substitute the switch expression. The problem is that refmt will always reorder comments placed directly after a switch expression, making it impossible to get the boundaries of the region right unless autoformat isn't used on the file.

Reply:
The approach with parenthesis sounds good to me. Another alternative I was thinking about is to just inline these problematic snippets in the markdown... If there are not too many of them, then it should be ok.

Reply:
> inline these problematic snippets in the markdown

It's likely I'll end up doing that in the future, but this refmt issue applies to all the snippets in this particular file :see_no_evil: